worlds/sm64ex/Rules.py

```diff
@@ -16,8 +16,13 @@ def set_rules(world, player: int, area_connections):
     if world.AreaRandomizer[player].value >= 1: # Some randomization is happening, randomize Courses
         entrance_ids = list(range(len(sm64paintings)))
         world.random.shuffle(entrance_ids)
-        entrance_ids = entrance_ids + list(range(len(sm64paintings), len(sm64paintings) + len(sm64secrets)))
-    if world.AreaRandomizer[player].value == 2: # Secret Regions as well
+        if world.AreaRandomizer[player].value == 3: # Shuffle Secret Regions only among themselves
+            secret_entrance_ids = list(range(len(sm64paintings), len(sm64paintings) + len(sm64secrets)))
+            world.random.shuffle(secret_entrance_ids)
+            entrance_ids += secret_entrance_ids
+        else:
+            entrance_ids += list(range(len(sm64paintings), len(sm64paintings) + len(sm64secrets)))
+    if world.AreaRandomizer[player].value == 2: # Shuffle Secret Regions with courses
```

Comment:
I'd prefer instead:
```
if world.AreaRandomizer[player].value >= 2: // Randomize Secrets as well
    secret_entrance_ids = list(range(len(sm64paintings), len(sm64paintings) + len(sm64secrets)))
    world.random.shuffle(secret_entrance_ids)
    entrance_ids += secret_entrance_ids
if world.AreaRandomizer[player].value == 3: // Randomize Courses and Secrets in one pool
    // The rest
```
This will cause the list to be shuffled one extra time, but i think it looks cleaner

Reply:
I updated it a little differently, let me know what you think

Reply:
Sorry, but I'd still prefer my way. While yours avoids the double shuffle, it becomes less readable due to nested branch. Later on when someone may want to add more settings this could become annoying

Reply:
you've got secret entrance IDs only added to the list if arearandomizer >= 2

What do you think about this?

```    secret_entrance_ids = list(range(len(sm64paintings), len(sm64paintings) + len(sm64secrets)))
    course_entrance_ids = list(range(len(sm64paintings)))
    if world.AreaRandomizer[player].value >= 1:  # Some randomization is happening, randomize Courses
        world.random.shuffle(course_entrance_ids)
    if world.AreaRandomizer[player].value == 2:  # Randomize Secrets as well
        world.random.shuffle(secret_entrance_ids)
    entrance_ids = course_entrance_ids + secret_entrance_ids
    if world.AreaRandomizer[player].value == 3:  # Randomize Courses and Secrets in one pool
        world.random.shuffle(entrance_ids)
        # etc```

Reply:
Yeah this seems nice